src/System.Linq/src/System/Linq/Repeat.cs

```diff
@@ -99,6 +99,18 @@ public List<TResult> ToList()
                 return list;
             }
 
+            public HashSet<TResult> ToHashSet(IEqualityComparer<TResult> comparer)
+            {
+                HashSet<TResult> hashSet = new HashSet<TResult>(_count, comparer);
+
+                for (int i = 0; i != _count; ++i)
+                {
+                    hashSet.Add(_current);
+                }
```

Comment:
With a comparer like that, I'm inclined to consider any behaviour that doesn't physically cause the computer to go on fire as acceptable 😄